stan/math/rev/core/vari.hpp

```diff
@@ -210,9 +199,134 @@ class vari_view<T, require_all_t<bool_constant<!is_plain_type<T>::value>,
   template <typename S, typename K,
             require_convertible_t<S&, value_type>* = nullptr,
             require_convertible_t<K&, value_type>* = nullptr>
-  vari_view(const S& val, const K& adj) : val_(val), adj_(adj) {}
+  vari_view(S&& val, K&& adj) noexcept : val_(val), adj_(adj) {}
```

Comment:
`&&` should be paired with `std::forward` in 99% of the use cases. Also I am not sure we need perfect forwarding. This accepts Eigen expressions and not plain types. I think these do not use any dynamically allocated memory, so moving is no better than copying.

Reply:
Yeah I think this is just a lazy case from me which is why there's no forward here either.

Reply:
If they did hold dynamic memory I'm pretty sure that would be real bad for this design because that would mean it needs to inherit from chainable_alloc(). ftr this view type is assuming the expressions here don't alloc any dynamic memory

Reply:
That is fine.